### PR TITLE
[#90]: Add Windows PEAK PCAN Support

### PR DESCRIPTION
**Pull request by ad3154**
## It works on Windows now

* Adds the windows PCAN libraries from the PCAN API. 
* Adds a new hardware plugin for PCAN on Windows. (The linux one is different and I will need to add it separately). 
* Added readme for 3rd party drivers where we can put any important info related to them. In this case I just wanted a place to stick their license notice.
* Made all examples support Windows if compiled with -DCAN_DRIVER=WindowsPCANBasic. 
* Fixed attempting to compile unit tests if BUILD_TESTING is enabled but socket CAN is not present. 
* Added a bunch of crazy CMake to detect the target architecture when compiling with MSVC because that functionality is totally broken in CMAKE. See https://gitlab.kitware.com/cmake/cmake/-/issues/15170

### CODE REVIEW

File: hardware_integration/src/pcan_basic_windows_plugin.cpp [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+bool PCANBasicWindowsPlugin::read_frame(isobus::HardwareInterfaceCANFrame &canFrame)
+{
+	TPCANStatus result;
+	TPCANMsg CANMsg;
+	TPCANTimestamp CANTimeStamp;
+	bool retVal = false;
+
+	result = CAN_Read(handle, &CANMsg, &CANTimeStamp);
+
+	if (result == PCAN_ERROR_OK)
```
Comment (GwnDaan) [inline hardware_integration/src/pcan_basic_windows_plugin.cpp:56]: PCAN_ERROR_OK == result

**Review by GwnDaan — COMMENTED** [review]
Good job! Having hardware support for windows can usb devices allows for much faster and easier iteration of dev builds. I have a few minor style related issues but further than that looks good to me.

File: hardware_integration/CMakeLists.txt [hunk 2]
```diff
@@ -8,22 +8,29 @@ set(HARDWARE_INTEGRATION_SRC_DIR "src")
 set(HARDWARE_INTEGRATION_INCLUDE_DIR "include/isobus/hardware_integration")
 
 if(NOT CAN_DRIVER)
-  message(AUTHOR_WARNING "No CAN driver specified, choosing Linux socket CAN by default. Use -DCAN_DRIVER=socketCAN to avoid this, or choose another driver and set it explicitly.")
-  set(CAN_DRIVER "socketCAN")
+    message(AUTHOR_WARNING "No CAN driver specified, choosing Linux socket CAN by default. Use -DCAN_DRIVER=\"SocketCAN\" to specify another driver. or -DCAN_DRIVER=\"SocketCAN;WindowsPCANBasic\" to specify multiple drivers.")
+	set(CAN_DRIVER "SocketCAN")
```
[the remark this answers is not in the document]
Comment (ad3154) [reply hardware_integration/CMakeLists.txt]: Ran `cmake-format` on all CMakeLists.txt files to normalize whitespace

File: CMakeLists.txt [hunk 3]
```diff
@@ -82,12 +82,14 @@ if(NOT TARGET GTest::gtest_main)
   add_library(GTest::gtest_main ALIAS GTest::Main)
 endif()
 
+if ("SocketCAN" IN_LIST CAN_DRIVER)
 add_executable(unit_tests test/address_claim_test.cpp test/test_CAN_glue.cpp test/identifier_tests.cpp test/dm_13_tests.cpp)
 target_link_libraries(unit_tests PRIVATE GTest::gtest_main ${PROJECT_NAME}::Isobus ${PROJECT_NAME}::HardwareIntegration ${PROJECT_NAME}::SystemTiming)
 
 include(GoogleTest)
 gtest_discover_tests(unit_tests name_tests identifier_tests)
 endif()
```
[the remark this answers is not in the document]
Comment (ad3154) [reply CMakeLists.txt]: Ran `cmake-format` on all CMakeLists.txt files to normalize whitespace

**Pull request merged**